src/App.tsx

```diff
@@ -14,11 +16,110 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [sortGoods, setSortGoodsBy] = useState('');
+  const [goodsLength, setGoodsLength] = useState(1);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortByAlphabet = () => {
+    setSortGoodsBy('Alphabet');
+  };
+
+  const sortByLength = () => {
+    setSortGoodsBy('Length');
+  };
+
+  const reset = () => {
+    setGoodsLength(1);
+    setIsReversed(false);
+    setSortGoodsBy('');
+  };
+
+  const reverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const goodsList = useMemo(() => {
+    return [...goodsFromServer]
```

Comment:
filter is not mutating method

```suggestion
    return goodsFromServer
```